### PR TITLE
Simplified version of "ecosystem in a box"

### PR DESCRIPTION
**Pull request by CapCap**
## Motivation
 Add functionality for #7477 - simplified version. Easy instructions to launch a validator/faucet, and then connect to it via cli client.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan
run docker composes. Minting via cli works

### CODE REVIEW

File: docker/compose/README.md
Comment (bmwill) [inline docker/compose/README.md:25]: I suppose we don't need waypoints anymore? @davidiw I'm not as familiar with waypoints as they are today so maybe you have some comments?

Comment (bmwill) [reply docker/compose/README.md:25]: If i'm reading correctly below, the validator will populate all of these things in a shared volume and the other docker images mount that shared volume to get access to these things. Its nice that we're able to stop needing these files checked into the repo.

**Review by bmwill — COMMENTED** [review]
I think this generally looks much cleaner, I'll give @davidiw a chance to do a quick pass over this before we land it though.

**Comment by bmwill** [comment]
Oh one other thing, you should take a look at the [contributing](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#authoring-clean-commits) guide for some guidance on how to structure your commits.

**Review by rexhoffman — COMMENTED** [review]
I'd recommend merging the two docker-compose files in to one given the shared resources and have the default action to be start the entire cluster, developers could ask docker hub to deploy one or the other service if they want via compose's command line then.

**Comment by rexhoffman** [comment]
Also CI currently tests nothing around docker compose, it's possible to setup a gha job on a host vm to run docker-compose and fire off a few transactions.

**Comment by bmwill** [comment]
> Also CI currently tests nothing around docker compose, it's possible to setup a gha job on a host vm to run docker-compose and fire off a few transactions.

That would be an awesome addition to make sure this doesn't become stale

File: docker/compose/README.md
Comment (xli) [reply docker/compose/README.md:25]: we don't need waypoints in general for non-verifying client.
waypoint can be used for validating server latest blocks, but will need a verifying client to handle the logic.
so for starting a test env in docker, there is no need.

Comment (bmwill) [reply docker/compose/README.md:25]: I think that the `cli` still ends up using the waypoint for one reason or another.

Comment (CapCap) [reply docker/compose/README.md:25]: even though there might not be a need for it, the cli currently requires it: 
```
error: The following required arguments were not provided:
     --waypoint <waypoint>
     --waypoint_url <waypoint_url>
```

in either case the waypoint does get populated by the validator and exists on the shared volume (for whatever might need it)

**Comment by CapCap** [comment]
Added a test script which can be run in CI :-)

**Comment by bmwill** [comment]
> Added a test script which can be run in CI :-)
Awesome!

@rexhoffman how would you like to integrate this into the CI pipeline?

**Comment by CapCap** [comment]
@rexhoffman I've added an attempt which builds the validator+client images and runs them through the test- I stuck it in ci-test as it'd be great for these to be blocking to guarantee the docker-compose are in a good state (as this one is going to be used in the other repos, too)

File: .github/workflows/ci-test.yml
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:256]: this job needs to be referenced in the last job in this file to gate changes landing, follow the pattern there and you'll see this job visualized as input to the last job.   It'd be good to see it succeed/fail once as well.

File: docker/compose/README.md
Comment (davidiw) [reply docker/compose/README.md:25]: Once you get the Rust SDK done, we should build a new CLI....

Comment (davidiw) [reply docker/compose/README.md:25]: The Faucet doesn't use the CLI any more afaiu, the old one did.

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (davidiw) [inline docker/compose/validator-testnet/docker-compose.yaml:1]: might recommend we have a single node testnet docker and a more complete ecosystem.

File: scripts/test_docker_compose.sh
Comment (davidiw) [inline scripts/test_docker_compose.sh:1]: hmm... should we put this into docker/compose/test.sh ?

**Review by davidiw — COMMENTED** [review]
coming together nicely!

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (CapCap) [reply docker/compose/validator-testnet/docker-compose.yaml:1]: have gone back and forth on that with @xli- general consensus so far has been generally if you're launching a testnet you want a faucet: in the event you don't, you can always ignore it or change the `docker-compose` command. In the spirit of keeping things simple, I would advocate for keeping them together as is and potentially creating additional `docker-compose` if we have the specific need. How does that sound?

File: docker/compose/README.md
Comment (CapCap) [reply docker/compose/README.md:25]: the faucet doesn't use the CLI, but the CLI does use the faucet (I think I'm missing something)

File: scripts/test_docker_compose.sh
Comment (CapCap) [reply scripts/test_docker_compose.sh:1]: ha that's where I had it originally :-)

File: .github/workflows/ci-test.yml
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:419]: not good enough, look at the job itself.   you should echo out the status, and add it to the big boolean && expression below.

Comment (CapCap) [reply .github/workflows/ci-test.yml:419]: oi completely missed it, sorry about that

File: docker/compose/README.md
Comment (xli) [inline docker/compose/README.md:9]: I think this client means a cli right? maybe we need name it client-cli or cli

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (xli) [inline docker/compose/validator-testnet/docker-compose.yaml:58]: you may change this to "http://validator:8080/v1", then we don't need ip address setting for validator

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (xli) [inline docker/compose/validator-testnet/docker-compose.yaml:39]: above 3 is not used, you can see we set them up for faucet command by arguments already.

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (xli) [inline docker/compose/validator-testnet/docker-compose.yaml:42]: I think we don't need this ip address setting, any reason we need it?

File: docker/compose/README.md
Comment (CapCap) [reply docker/compose/README.md]: doc updated

File: docker/compose/client/docker-compose.yaml
Comment (xli) [inline docker/compose/client/docker-compose.yaml:1]: may rename dir from client to client-cli too

File: docker/compose/README.md
Comment (xli) [inline docker/compose/README.md:28]: client -> client-cli

File: scripts/test_docker_compose.sh
Comment (xli) [inline scripts/test_docker_compose.sh:36]: client -> client-cli

File: docker/compose/README.md
Comment (xli) [inline docker/compose/README.md:9]: it seems this rename triggers a lot renaming :(, if you want to keep original "client", I'm fine with it.

**Review by xli — APPROVED** [review]

File: docker/compose/README.md
Comment (davidiw) [reply docker/compose/README.md:25]: CLI does require a waypoint but it really doesn't need it (from an internal perspective).

File: docker/compose/validator-testnet/docker-compose.yaml
Comment (davidiw) [reply docker/compose/validator-testnet/docker-compose.yaml:1]: Hmm... I guess if we run this all in the background and still export the root key, it is fine either way.

File: docker/compose/README.md
Comment (CapCap) [reply docker/compose/README.md:9]: wasn't too bad, I agree it provides clarity :-)

File: docker/compose/README.md
Comment (CapCap) [reply docker/compose/README.md:25]: @davidiw I'll put up a PR after this to remove the requirement we pass one (I've been taking rust tutorials and am excited to see if I've learned anything :-P)

**Pull request merged**
